lib/sarif/semgrep_sarif.rb

```diff
@@ -0,0 +1,79 @@
+require 'salus/bugsnag'
+
+module Sarif
+  class SemgrepSarif < BaseSarif
+    include Salus::SalusBugsnag
+
+    SEMGREP_URI = 'https://semgrep.dev/'.freeze
+
+    def initialize(scan_report)
+      super(scan_report)
+      @uri = SEMGREP_URI
+      @logs = parse_scan_report!
+      @issues = Set.new
+    end
+
+    def parse_scan_report!
+      hits = @scan_report.to_h.dig(:info, :hits)
+      warnings = @scan_report.to_h.dig(:warn, :semgrep_non_fatal) || []
+      hits.concat(warnings)
+    end
+
+    def parse_issue(issue)
+      if issue.key?(:type)
+        parse_warning(issue)
+      else
+        parse_hit(issue)
+      end
+    end
+
+    def parse_hit(hit)
+      id = hit[:hit] || hit[:msg]
+      return nil if @issues.include?(id)
+
+      @issues.add(id)
+      location = hit[:hit].split(":") # [file_name, line, code_preview]
+      {
+        id: hit[:pattern],
+        name: "#{hit[:pattern]} / #{hit[:msg]}",
+        level: "HIGH",
+        details: "Pattern: #{hit[:pattern]}\nMessage:#{hit[:msg]}\nForbidden:#{hit[:forbidden]}"\
+        "\nRequired:#{hit[:required]}",
```

Comment:
I wonder if we need the forbidden and required fields... that may be additional context we can drop here.  Thoughts?